apps/api/src/hyp3_api/validation.py

```diff
@@ -241,6 +245,37 @@ def check_opera_rtc_s1_static_coverage(job: dict, _) -> None:
         )
 
 
+def check_aria_s1_gunw_dates(job: dict, _) -> None:
+    def get_date_from_job(date_param_key: str) -> date:
+        param_date_str = job['job_parameters'][date_param_key]
+        return datetime.strptime(param_date_str, '%Y-%m-%d').date()
+
+    job_dates = {
+        'reference_date': get_date_from_job('reference_date'),
+        'secondary_date': get_date_from_job('secondary_date'),
+    }
+
+    s1_start_date = date(2014, 6, 15)
+    todays_date = date.today()
+
+    for job_date_key, job_date in job_dates.items():
+        if job_date > todays_date:
+            raise DateValidationError(f'"{job_date_key}" is {job_date} which is a value in the future.')
+
+        if job_date < s1_start_date:
+            raise DateValidationError(
+                f'"{job_date_key}" is {job_date} which is before the start of the sentinel 1 mission ({s1_start_date}).'
+            )
+
+    if job_dates['reference_date'] == job_dates['secondary_date']:
+        raise DateValidationError(
+            f'reference and secondary dates are equal, must be different dates ({job_dates["reference_date"]}).'
+        )
+
+    if job_dates['reference_date'] > job_dates['secondary_date']:
+        raise DateValidationError('reference date must be earlier than secondary date.')
```

Comment:
We want reference to be the later date, and there's no need for a separate equality check:
```suggestion
    if not job_dates['secondary_date'] < job_dates['reference_date']:
        raise DateValidationError('secondary date must be earlier than reference date.')
```